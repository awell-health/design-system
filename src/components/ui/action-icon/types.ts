export enum ActionType {
  ApiCall = 'API_CALL',
  ApiCallGraphql = 'API_CALL_GRAPHQL',
  Calculation = 'CALCULATION',
  Checklist = 'CHECKLIST',
  ClinicalNote = 'CLINICAL_NOTE',
  Form = 'FORM',
  Message = 'MESSAGE',
  Plugin = 'PLUGIN',
  PushToEmr = 'PUSH_TO_EMR'
}
