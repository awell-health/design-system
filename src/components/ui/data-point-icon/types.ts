export type DataPointValueType =
  | 'attachment'
  | 'attachments_array'
  | 'boolean'
  | 'date'
  | 'json'
  | 'number'
  | 'numbers_array'
  | 'string'
  | 'strings_array'
  | 'telephone';
