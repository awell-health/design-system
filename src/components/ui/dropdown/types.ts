export type DropdownItem = {
  label: string | JSX.Element;
  onClick: () => void;
};
