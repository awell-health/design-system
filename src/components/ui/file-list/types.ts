export interface FileListItem {
  name: string;
  id?: string;
  size?: number;
  type?: string;
  progress?: number;
  error?: string;
}
