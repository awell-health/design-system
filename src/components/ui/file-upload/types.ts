export type AcceptedFileTypes = '.jpg' | '.jpeg' | '.png' | '.pdf' | '.doc' | '.docx' | '.txt';
