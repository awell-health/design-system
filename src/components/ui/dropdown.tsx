import { cn } from "@/lib/utils";

export type DropdownItem = {
  label: string
  onClick: () => void
};

export interface DropdownProps {
  buttonLabel: string | JSX.Element
  items: Array<DropdownItem>
  buttonClassNames?: string
  itemClassNames?: string
}

function Dropdown(props: DropdownProps) {
  const { items, buttonLabel, buttonClassNames, itemClassNames } = props;

  const renderItem = ({ label, onClick }: DropdownItem): JSX.Element => (
    <li className={cn(itemClassNames)}>
      <a onClick={onClick}>{label}</a>
    </li>
  );

  return (
    <div className="dropdown">
      <div tabIndex={0} role="button" className={cn("btn m-1", buttonClassNames)}>
        {buttonLabel}
      </div>
      <ul
        tabIndex={0}
        className="dropdown-content z-[1] menu p-2 shadow bg-base-100 rounded-box w-52"
      >
        {items.map(renderItem)}
      </ul>
    </div>
  );
}

export { Dropdown };
