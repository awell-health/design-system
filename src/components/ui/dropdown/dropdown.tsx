import { cva, type VariantProps } from 'class-variance-authority';
import { cn } from '@/lib/utils';
import { DropdownItem } from './types';

const dropdownVariants = cva('dropdown', {
  variants: {
    placement: {
      default: '',
      top: 'dropdown-top dropdown-start',
      left: 'dropdown-left',
      right: 'dropdown-right',
      bottom: 'dropdown-bottom dropdown-start',
      'bottom-end': 'dropdown-bottom dropdown-end',
      'top-end': 'dropdown-top dropdown-end'
    }
  },
  defaultVariants: {
    placement: 'default'
  }
});

export interface DropdownProps
  extends React.ButtonHTMLAttributes<HTMLDivElement>,
    VariantProps<typeof dropdownVariants> {
  buttonLabel: string | JSX.Element;
  items: Array<DropdownItem>;
  buttonClassNames?: string;
  itemClassNames?: string;
}

function Dropdown(props: DropdownProps) {
  const { items, buttonLabel, buttonClassNames, itemClassNames, placement, className } = props;

  const renderItem = ({ label, onClick }: DropdownItem, i: number): JSX.Element => (
    <li key={i} className={cn(itemClassNames)}>
      <a onClick={onClick}>{label}</a>
    </li>
  );

  return (
    <div className={cn(dropdownVariants({ placement, className }))}>
      <div tabIndex={0} role='button' className={cn('btn m-1', buttonClassNames)}>
        {buttonLabel}
      </div>
      <ul
        tabIndex={0}
        className='dropdown-content z-[1] menu p-2 shadow bg-base-100 rounded-box w-52'
      >
        {items.map(renderItem)}
      </ul>
    </div>
  );
}

export { Dropdown };
