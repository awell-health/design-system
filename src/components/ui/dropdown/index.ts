import { Dropdown } from './dropdown';
import { DropdownItem } from './types';

export { Dropdown, type DropdownItem };
