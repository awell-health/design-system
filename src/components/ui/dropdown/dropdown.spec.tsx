import { fireEvent, render } from '@testing-library/react';
import { expect, it, describe, vi } from 'vitest';
import { Dropdown } from './dropdown';
import { DropdownItem } from './types';

describe('Dropdown', () => {
  const clickMock = vi.fn();

  const items: DropdownItem[] = [
    { label: 'Test 1', onClick: clickMock },
    {
      label: <span>JSX item</span>,
      onClick: clickMock
    }
  ];

  it('match snapshot', () => {
    const result = render(<Dropdown items={items} buttonLabel={'Button Label'} />);

    expect(result).toMatchSnapshot();
  });

  it('triggers item click event', () => {
    const { getByText } = render(<Dropdown items={items} buttonLabel={'Button Label'} />);

    fireEvent.click(getByText('Test 1'));

    expect(clickMock).toHaveBeenCalledOnce();
  });
});
