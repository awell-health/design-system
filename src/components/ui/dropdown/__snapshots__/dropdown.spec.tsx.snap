// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dropdown > match snapshot 1`] = `
<div>
  <div
    class="dropdown"
  >
    <div
      class="btn m-1"
      role="button"
      tabindex="0"
    >
      Button Label
    </div>
    <ul
      class="dropdown-content z-[9999] menu p-2 shadow bg-base-100 rounded-box w-52"
      tabindex="0"
    >
      <li
        class=""
      >
        <a
          class=""
        >
          Test 1
        </a>
      </li>
      <li
        class=""
      >
        <a
          class=""
        >
          <span>
            JSX item
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
